### PR TITLE
Continues to add default sql indexes and table optimisations

### PR DESCRIPTION
**Pull request by Jordie0608**
See #24345 and #24277 since I really shouldn't be messing up upstream branches in the first place.
@MrStonedOne

### CODE REVIEW

File: SQL/optimisations_2017-02-19.sql [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+, CHANGE COLUMN `unbanned_ipTEMP` `unbanned_ip` INT(10) UNSIGNED NULL DEFAULT NULL;
+COMMIT;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`connection_log`
+ ADD COLUMN `server_ip` INT UNSIGNED NOT NULL AFTER `serverip`
+, ADD COLUMN `server_port` SMALLINT UNSIGNED NOT NULL AFTER `server_ip`
+, ADD COLUMN `ipTEMP` INT UNSIGNED NOT NULL AFTER `ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`ban`
```
Comment (MrStonedOne) [inline SQL/optimisations_2017-02-19.sql]: ?

File: SQL/optimisations_2017-02-19.sql [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+, CHANGE COLUMN `lakey` `lakey` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `gender` `gender` ENUM('neuter', 'male', 'female', 'plural') NOT NULL
+, CHANGE COLUMN `bruteloss` `bruteloss` SMALLINT UNSIGNED NOT NULL
+, CHANGE COLUMN `brainloss` `brainloss` SMALLINT UNSIGNED NOT NULL
+, CHANGE COLUMN `fireloss` `fireloss` SMALLINT UNSIGNED NOT NULL
+, CHANGE COLUMN `oxyloss` `oxyloss` SMALLINT UNSIGNED NOT NULL
+, ADD COLUMN `server_ip` INT UNSIGNED NOT NULL AFTER `server`
+, ADD COLUMN `server_port` SMALLINT UNSIGNED NOT NULL AFTER `server_ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`ban`
```
Comment (MrStonedOne) [inline SQL/optimisations_2017-02-19.sql]: ?

File: SQL/optimisations_2017-02-19.sql [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+, CHANGE COLUMN `createdby_ip` `createdby_ip` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `for_trialmin` `for_trialmin` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `dontshow` `dontshow` TINYINT(1) UNSIGNED NOT NULL;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`poll_textreply`
+ CHANGE COLUMN `replytext` `replytext` VARCHAR(2048) NOT NULL
+, ADD COLUMN `ipTEMP` INT UNSIGNED NOT NULL AFTER `ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`ban`
```
Comment (MrStonedOne) [inline SQL/optimisations_2017-02-19.sql]: ?

File: SQL/optimisations_2017-02-19.sql [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+ DROP COLUMN `ip`
+, CHANGE COLUMN `ipTEMP` `ip` INT(10) UNSIGNED NOT NULL;
+COMMIT;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`poll_vote`
+ CHANGE COLUMN `ckey` `ckey` VARCHAR(32) NOT NULL
+, ADD COLUMN `ipTEMP` INT UNSIGNED NOT NULL AFTER `ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`ban`
```
Comment (MrStonedOne) [inline SQL/optimisations_2017-02-19.sql]: ?

**Review by MrsTonedOne — CHANGES_REQUESTED** [review]

File: SQL/optimisations_2017-02-19.sql [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+, CHANGE COLUMN `secret` `secret` TINYINT(1) UNSIGNED NOT NULL;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`player`
+ ADD COLUMN `ipTEMP` INT UNSIGNED NOT NULL AFTER `ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`player`
+ SET `ip` = IF(`ip` IS NULL OR `ip` = '','0',`ip`);
+UPDATE `feedback`.`player`
+ SET `ipTEMP` = INET_ATON(`ip`);
```
Comment (MrStonedOne) [inline SQL/optimisations_2017-02-19.sql]: Should maybe use the single query method you figured out for the ips in other tables on this one too.

File: SQL/optimisations_2017-02-19.sql [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+COMMIT;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`death`
+ CHANGE COLUMN `pod` `pod` VARCHAR(50) NOT NULL
+, CHANGE COLUMN `coord` `coord` VARCHAR(32) NOT NULL
+, CHANGE COLUMN `mapname` `mapname` VARCHAR(32) NOT NULL
+, CHANGE COLUMN `job` `job` VARCHAR(32) NOT NULL
+, CHANGE COLUMN `special` `special` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `name` `name` VARCHAR(32) NOT NULL
```
Comment (MrStonedOne) [inline SQL/optimisations_2017-02-19.sql]: `#1406 - Data too long for column 'name' at row 3570`

Comment (MrStonedOne) [inline SQL/optimisations_2017-02-19.sql]: longest name is 67 characters.

Comment (Jordie0608) [inline SQL/optimisations_2017-02-19.sql]: i blame badmins

**Pull request merged**